### PR TITLE
Feature/add onlyLatestVersions parameter to use cases

### PR DESCRIPTION
**Pull request by Milena-Czierlinski**

### CODE REVIEW

File: test/lib/testUtils.ts [hunk 1]
```diff
@@ -74,23 +79,23 @@ export async function syncUntilHasMessages(transportServices: TransportServices,
 
 export async function syncUntilHasMessageWithRequest(transportServices: TransportServices, requestId: string | CoreId): Promise<MessageDTO> {
     const filterRequestMessagesByRequestId = (syncResult: SyncEverythingResponse) => {
-        return syncResult.messages.filter((m) => m.content["@type"] === "Request" && m.content.id === requestId);
+        return syncResult.messages.filter((m) => m.content["@type"] === "Request" && (m.content.id === requestId || m.content.id === requestId.toString()));
```
Comment (stnmtz) [inline test/lib/testUtils.ts:82]: Could you explain why this change is neccessary? It seems to me, that the SyncResult's content here might not be a DTO (with id as strings) but an object (with an CoreId as id) which would be a bug.

Comment (jkoenig134) [reply test/lib/testUtils.ts:82]: I think this can just be written as 
```ts
        return syncResult.messages.filter((m) => m.content["@type"] === "Request" && m.content.id === requestId.toString());
```
as you can also run toString on a string :)

Comment (Milena-Czierlinski) [reply test/lib/testUtils.ts:82]: `typeof syncResult.messages[0].content` -> object
`typeof syncResult.messages[0].content.id` -> string

But we allow `notificationId` to be either a string or a CoreId. Thus, in the latter case we need to convert it to a string to be able to compare it to the `syncResult.messages[0].content.id`.

File: package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
         "test:ci:lokijs": "USE_LOKIJS=true jest -i",
         "test:ci:mongodb": "CONNECTION_STRING='mongodb://127.0.0.1:27017' jest -i",
         "test:local": "npm run test:local:mongodb && npm run test:local:lokijs",
-        "test:local:ferretdb": ". scripts/testSetupFerret.sh && jest -w 3",
+        "test:local:ferretdb": ". scripts/testSetupFerret.sh && jest -w 1",
```
Comment (jkoenig134) [inline package.json:36]: can you remove the `-w` flag instead? I think we only have 1 worker configured.

File: src/useCases/consumption/attributes/GetOwnSharedAttributes.ts [hunk 3]
```diff
@@ -48,6 +56,10 @@ export class GetOwnSharedAttributesUseCase extends UseCase<GetOwnSharedAttribute
         const flattenedQuery = flattenObject(query);
         const dbQuery = GetAttributesUseCase.queryTranslator.parse(flattenedQuery);
 
+        if (typeof request.onlyLatestVersions === "undefined" || request.onlyLatestVersions) {
```
Comment (jkoenig134) [inline src/useCases/consumption/attributes/GetOwnSharedAttributes.ts:59]: you can simplify this to 
```
        if (request.onlyLatestVersions ?? false) {
```
also in the other files

Comment (jkoenig134) [reply src/useCases/consumption/attributes/GetOwnSharedAttributes.ts]: as discussed I assumed a wrong default.
should be `request.onlyLatestVersions ?? true`

File: test/lib/testUtils.ts [hunk 4]
```diff
@@ -55,7 +60,7 @@ export async function syncUntil(transportServices: TransportServices, until: (sy
 
         iterationNumber++;
         criteriaMet = until(finalSyncResult);
-    } while (!criteriaMet && iterationNumber <= 10);
+    } while (!criteriaMet && iterationNumber <= 30);
```
Comment (jkoenig134) [inline test/lib/testUtils.ts:63]: this should not be nessecary, I think I screwed up the backbone action.

Comment (jkoenig134) [reply test/lib/testUtils.ts]: or like I see you matched the wrong parameter :D so please downgade again :D

Comment (Milena-Czierlinski) [reply test/lib/testUtils.ts]: Nevertheless, it was quite unstable on my local setup.

Comment (jkoenig134) [reply test/lib/testUtils.ts]: yeah - we have to fix the backbone setup

**Review by jkoenig134 — APPROVED** [review]
